alcs-frontend/src/app/services/planning-review/planning-referral.service.spec.ts

```diff
@@ -0,0 +1,153 @@
+import { HttpClient } from '@angular/common/http';
+import { TestBed } from '@angular/core/testing';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { of, throwError } from 'rxjs';
+import { ToastService } from '../toast/toast.service';
+import { PlanningReferralService } from './planning-referral.service';
+
+describe('PlanningReferralService', () => {
+  let service: PlanningReferralService;
+  let httpClient: DeepMocked<HttpClient>;
+  let toastService: DeepMocked<ToastService>;
+
+  beforeEach(() => {
+    httpClient = createMock();
+    toastService = createMock();
+
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: HttpClient,
+          useValue: httpClient,
+        },
+        {
+          provide: ToastService,
+          useValue: toastService,
+        },
+      ],
+    });
+    service = TestBed.inject(PlanningReferralService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should call get for fetchByCardUuid', async () => {
+    httpClient.get.mockReturnValue(
+      of({
+        fileNumber: '1',
+      }),
+    );
+
+    await service.fetchByCardUuid('card-uuid');
+
+    expect(httpClient.get).toHaveBeenCalledTimes(1);
+  });
+
+  it('should show an error toast message if fetchByCardUuid fails', async () => {
+    httpClient.get.mockReturnValue(
+      throwError(() => {
+        new Error('');
+      }),
+    );
+
+    const res = await service.fetchByCardUuid('card-uuid');
+
+    expect(httpClient.get).toHaveBeenCalledTimes(1);
+    expect(res).toBeUndefined();
```

Comment:
This one's kind of the opposite problem. Shouldn't we only be checking for the toast?